### PR TITLE
mgr: missing lock release in DaemonServer::handle_report()

### PR DESCRIPTION
**Pull request by vshankar**

### CODE REVIEW

File: src/mgr/DaemonServer.cc [hunk 1]
```diff
@@ -562,6 +562,7 @@ bool DaemonServer::handle_report(const ref_t<MMgrReport>& m)
       auto priv = m->get_connection()->get_priv();
       auto session = static_cast<MgrSession*>(priv.get());
       if (!session) {
+        lock.unlock();
```
Comment (tchaikov) [inline src/mgr/DaemonServer.cc]: could you take this opportunity to use a `unique_lock<>` ?

Comment (vshankar) [reply src/mgr/DaemonServer.cc]: ack

File: src/mgr/DaemonServer.cc [hunk 2]
```diff
@@ -505,15 +505,15 @@ bool DaemonServer::handle_report(const ref_t<MMgrReport>& m)
 
 
   {
-    lock.lock();
+    std::unique_lock locker(lock);
```
Comment (batrick) [inline src/mgr/DaemonServer.cc]: minor nit: better would be `std::scoped_lock`

Comment (vshankar) [reply src/mgr/DaemonServer.cc]: yep

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-10-09_15:12:02-rados-wip-kefu-testing-2019-10-09-1243-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2019-10-14_06:20:56-rados-wip-kefu-testing-2019-10-14-1111-distro-basic-smithi/

failures tracked by 

- https://tracker.ceph.com/issues/42301
- https://tracker.ceph.com/issues/40026

**Pull request merged**